pkg/telemetry/histogram.go

```diff
@@ -0,0 +1,52 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Histogram tracks the value of one health metric of the Agent.
+type Histogram interface {
+	// Observe  the value to the Gauge value.
+	Observe(value float64, tagsValue ...string)
+	// Delete deletes the value for the Gauge with the given tags.
+	Delete(tagsValue ...string)
+}
+
+// NewGauge creates a Gauge with default options for telemetry purpose.
+// Current implementation used: Prometheus Gauge
```

Comment:
comment needs an update (applies to a lot of other comments in your changes to `telemetry` that still refer to Gauge)